packages/server/__tests__/controllers/guilds.test.js

```diff
@@ -0,0 +1,201 @@
+      new PrismaClientKnownRequestError("record not found", { code: "P2025" })
+    );
+
+    await expect(
+      GuildsController.getGuild("e2bdade9-4bf2-4220-8020-e09266363762")
+    ).rejects.toThrow(PrismaClientKnownRequestError);
+  });
+
+  /*
+  Unit Tests: Creating Guilds
+  - Successfully creating guild ✅
+  - Trying to create a guild without the required fields (should throw error) ✅
+  NOTE: use objectContaining to test for specific fields in the guildObject ✅
+  */
+
+  // Define Test Data For Create
+  const testCreateGuild = {
+    name: "Software Engineering Association",
+    handler: "sea",
+    description: "test description",
+    category: "software engineering",
+  };
+
+  const testinvalidCreateGuild = {
+    location: "San Diego",
+    website: "apple.com",
+    banner: "apple.jpeg",
+    icon: "appleicon.jpeg",
+  };
+
+  test("should sucessfully create new guild", async () => {
+    prismaMock.guilds.create.mockResolvedValue(testCreateGuild);
+
+    await expect(
+      GuildsController.createGuild(testCreateGuild)
+    ).resolves.toEqual(expect.objectContaining(testCreateGuild));
+  });
+
+  test("should fail guild creation due to missing required fields", async () => {
+    prismaMock.guilds.create.mockResolvedValue(
+      new PrismaClientKnownRequestError("Missing required field error", {
+        code: "P2012",
+      })
+    );
+
+    await expect(
+      GuildsController.createGuild(testinvalidCreateGuild)
+    ).resolves.toThrow(PrismaClientKnownRequestError);
+  });
+
+  /*
+  Unit Tests: Updating Guilds
+  - Successfully updating multiple guild properties ✅
+  - Successfully updating one guild property ✅
+  - Trying to update a noneixstent guild property (should throw error)
+  - Trying to update a nonexistent guild (should throw error)
+  */
+
+  // Define Test Data For Update
+  const initialMultipleUpdateGuildResponse = {
+    name: "Apple",
+    handler: "Tim Cook",
+    description: "Come buy our overpriced iphones.",
+    category: "Guild Category",
+    location: "Optional Location",
+    website: "Optional Website",
+    tags: ["Tag1", "Tag2", "apple 15"],
+    banner: "URL to Banner Image",
+    icon: "URL to Icon Image",
+    media: ["Media1", "Media2", "Media3"],
+    isInviteOnly: true,
+  };
+
+  const expectedUpdateGuildResponse = {
+    guildId: "308b5bc4-7771-4514-99b7-9611cefb7e1d",
+    name: "Apple",
+    handler: "Tim Cook",
+    description: "Come buy our overpriced iphones.",
+    category: "Guild Category",
+    location: "Optional Location",
+    website: "Optional Website",
+    tags: ["Tag1", "Tag2", "apple 15"],
+    banner: "URL to Banner Image",
+    icon: "URL to Icon Image",
+    media: ["Media1", "Media2", "Media3"],
+    isInviteOnly: true,
+  };
+
+  const initialMultipleUpdateGuildResponseOneField = {
+    name: "Facebook",
+  };
+
+  const expectedUpdateGuildResponseOneField = {
+    guildId: "308b5bc4-7771-4514-99b7-9611cefb7e1d",
+    name: "Facebook",
+  };
```

Comment:
These variable names are starting to get really long. Since some of these are only used for one specific test, you can declare them inside each test block it is used in instead and keep the variable names shorter like `expected`.

For example for your "should update multiple guild properties successfully" test, something like this:
```js
test("should update multiple guild properties successfully", async () => {
  // this replaces expectedUpdateGuildResponse
  const expected = {
    guildId: "308b5bc4-7771-4514-99b7-9611cefb7e1d",
    name: "Apple",
    handler: "Tim Cook",
    description: "Come buy our iphones.",
    category: "Guild Category",
    location: "Optional Location",
    website: "Optional Website",
    tags: ["Tag1", "Tag2", "apple 15"],
    banner: "URL to Banner Image",
    icon: "URL to Icon Image",
    media: ["Media1", "Media2", "Media3"],
    isInviteOnly: true,
  };

  prismaMock.guilds.update.mockResolvedValue(expected);

  const result = await GuildsController.updateGuild(
    "308b5bc4-7771-4514-99b7-6666cefb7e1d",
    initialGuild // originally initialMultipleUpdateGuildResponse, replaced name because it was too long
  );

  expect(result).toEqual(
    expect.objectContaining(expectedUpdateGuildResponse)
  );
});
```